LeanMapperQuery/Entity.php

```diff
@@ -52,7 +54,29 @@ protected function queryProperty($field, IQuery $query)
 		if ($relationship instanceof Relationship\BelongsToMany) {
 			$targetTable = $relationship->getTargetTable();
 			$referencingColumn = $relationship->getColumnReferencingSourceTable();
-			$rows = $this->row->referencing($targetTable, $referencingColumn, new Filtering($filters), $relationship->getStrategy());
+			$strategy = $relationship->getStrategy();
+			$usesDetector = FALSE;
+
+			if ($strategy !== Result::STRATEGY_UNION) {
+				$usesDetector = TRUE;
+				$filters[] = function (Fluent $fluent) {
+					if ($fluent->_export('LIMIT') || $fluent->_export('OFFSET')) {
+						throw new InvalidStrategyException('Fluent uses LIMIT or OFFSET, use UNION strategy.');
+					}
+				};
+			}
+
+			$rows = [];
+
+			try {
+				$rows = $this->row->referencing($targetTable, $referencingColumn, new Filtering($filters), $strategy);
+
+			} catch (InvalidStrategyException $e) {
+				if ($usesDetector) {
```

Comment:
Teoreticky není třeba kontrolovat. Tuhle výjimku by němělo vyhodit nic jiného.

Reply:
Teoreticky, ale IMHO bude lepší ten filtr odebrat jen když je jistota, že jsme ho tam i přidali. Sice je nepravděpodobné, že by tu výjimku někdo vyhodil ze svého filtru, ale stát by se to mohlo.